src/components/Header/SpacesDropdown.tsx

```diff
@@ -1,11 +1,16 @@
-import { Collection, Space } from '@gitbook/api';
+'use client';
```

Comment:
The downside to having `use client` on this component is that the whole dropdown will no longer be a server component, even though it's only the individual menu items that need the current layout segment.

I'd create a new component called `SpacesDropdownItem`:

```
'use client';

export function SpacesDropdownItem(props: { space: Space; }) {
  const currentPathname = useSelectedLayoutSegment() ?? '';
  return <DropdownItem ... />
}
```

so we have Client Components further down the stack